e2etests/all_test.go

```diff
@@ -0,0 +1,135 @@
+// +build e2etests
+
+package e2etests
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"testing"
+
+	"github.com/gobuffalo/envy"
+	"github.com/stretchr/testify/suite"
+)
+
+type E2eSuite struct {
+	suite.Suite
+	goBinaryPath   string
+	env            []string
+	tmpDir         string
+	goModCache     string
+	sampleRepoPath string
+}
+
+type catalogRes struct {
+	Modules []struct {
+		Module  string `json:"module"`
+		Version string `json:"version"`
+	} `json:"modules"`
+}
+
+func (m *E2eSuite) SetupSuite() {
+	var err error
+	m.tmpDir, err = mkTemp()
+	if err != nil {
+		m.Fail("Failed to make temp dir", err)
+	}
+
+	m.goModCache = path.Join(m.tmpDir, "pkg", "mod")
+	m.sampleRepoPath = path.Join(m.tmpDir, "happy-path")
+	m.goBinaryPath = envy.Get("GO_BINARY_PATH", "go")
+
+	athensBin, err := buildAthens(m.goBinaryPath, m.tmpDir, m.env)
+	if err != nil {
+		m.Fail("Failed to build athens ", err)
+	}
+	stopAthens() // in case a dangling instance was around.
+	// ignoring error as if no athens is running it fails.
+	runAthensAndWait(athensBin, m.getEnv())
+	setupTestRepo(m.sampleRepoPath, "https://github.com/athens-artifacts/happy-path.git")
+}
+
+func (m *E2eSuite) TearDownSuite() {
+	err := stopAthens()
+	if err != nil {
+		m.Fail("Failed to stop athens", err)
+	}
+	chmodR(m.tmpDir, 0777)
```

Comment:
I think a cleaner way to do this is to just call clean modcache -- I think we should probably do it on the go_get_fetcher as well. But that's another topic :)

Reply:
I do agree to clear the cache instead of brutally erasing in `SetupTest`, as there we care only about the cache to be empty.
In `TearDown` though, I'd like to remove any additional folder that was created by the test suite, and I am afraid clearing the cache is not enough.

Reply:
@fedepaol right you can still remove the folder, but there won't be a need to `chmod` anything :)

Reply:
Yes I left it there as I did not remember why I added originally. I'll try to remove it and see if it works :-)